src/libvirtApi/domain.js

```diff
@@ -1054,3 +1056,15 @@ export async function domainReplaceSpice({ connectionName, id: objPath }) {
         throw ex; // not-covered: see above
     }
 }
+
+export async function domainAddTPM({ connectionName, vmName }) {
+    const args = ["virt-xml", "-c", `qemu:///${connectionName}`, "--add-device", "--tpm", "default,alias.name=tpm0", vmName];
```

Comment:
User specified alias names must start with `ua-`.  Since `tpm0` is considered a reserved alias name by libvirt it will be discarded when parsing the input XML.

https://libvirt.org/formatdomain.html#devices

There's no compelling reason to set the alias name manually, as the default auto-generated alias will be fine.

Reply:
Thanks @berrange ! I just closed the bugzilla. I'll drop this.

The only annoyance left is that this doesn't work on RHEL 8, but we can at least fix it for RHEL 9 and Fedora then.